src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -613,18 +612,16 @@ public IEnumerable<string> Audiences
             {
                 if (_audiences == null)
                 {
-                    _audiences = new List<string>();
-
-                    if (Payload.TryGetValue(JwtRegisteredClaimNames.Aud, out JsonElement audiences))
+                    lock (_audiences)
                     {
-                        if (audiences.ValueKind == JsonValueKind.String)
+                        if (_audiences == null)
                         {
-                            _audiences.Add(audiences.GetString());
-                        }
-                        else if (audiences.ValueKind == JsonValueKind.Array)
-                        {
-                            foreach (JsonElement jsonElement in audiences.EnumerateArray())
-                                _audiences.Add(jsonElement.ToString());
+                            _audiences = new List<string>();
```

Comment:
I believe you have a race condition here.

You need to wait until the collection is fully initialized before assigning the reference.

Reply:
@keegan-caruso nice catchfixed.